connector/src/yang/connector/grpc.py

```diff
@@ -0,0 +1,179 @@
+import logging
+import multiprocessing
+import re
+import socket
+import configparser
+import subprocess
+import psutil
+
+try:
+    from pyats.log.utils import banner
+    from pyats.connections import BaseConnection
+    from pyats.utils.secret_strings import to_plaintext
+    from pyats.easypy import runtime
+    from genie.libs.sdk.apis.utils import get_local_ip
+    from genie.libs.sdk.apis.iosxe.telemetry.configure import configure_telemetry_ietf_parameters
+    from unicon import Connection
+except ImportError:
+    # Standalone without pyats install
+    class BaseConnection:
+        class dev:
+            def __init__(self, dev_os):
+                self.os = dev_os
+
+        def __init__(self, device_os, **kwargs):
+            self.connection_info = {'protocol': 'gnmi'}
+            self._device = self.dev(device_os)
+            self.connection_info.update(kwargs)
+
+    class runtime:
+        import os
+        directory = os.curdir
+
+    def banner(string):
+        return string
+
+    def to_plaintext(string):
+        return string
+
+
+# create a logger for this module
+log = logging.getLogger(__name__)
+
+
+class Grpc(BaseConnection):
+    """Session handling for Grpc outbound connections.
+
+        Can be used with pyATS same as yang.connector.Netconf is used or
+        can be used as a standalone module.
+    """
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.device = kwargs.get('device')
+        dev_args = self.connection_info
+        self.log = log
+        self.log.setLevel(logging.INFO)
+
+        if protocol := dev_args.get('protocol', '').lower() != 'grpc':
+            msg = f"Invalid protocol {protocol}"
+            raise TypeError(msg)
+
+        self.username = dev_args.get('username', '')
+        self.password = dev_args.get('password', '')
+        if not self.username or not self.password:
+            credentials = dev_args.get('credentials', '')
+            if not credentials:
+                raise KeyError("No credentials found for testbed")
+            if 'grpc' not in credentials:
+                log.info(f"Credentials used from {next(iter(credentials))}")
+            grpc_uname_pwd = credentials.get('')
+            if not grpc_uname_pwd:
+                raise KeyError('No credentials found for gRPC testbed')
+
+        self.host = dev_args.get('host') or dev_args.get('ip')
+
+        self.transporter = dev_args.get('transporter', 'telegraf').lower()
+        self.output_file = dev_args.get('output_file', f'{runtime.directory}/mdt')
+        self.config_file = dev_args.get('config_file', None)
+        self.telemetry_subscription_id = dev_args.get('telemetry_subscription_id', 11172017)
+
+        self.telegraf_process = None
```

Comment:
Suggest to use a generic name so this code can be re-used.

```suggestion
        self.transport_process = None
```